sockeye/data_io.py

```diff
@@ -466,7 +530,43 @@ def load(self,
         return ParallelDataSet(data_source, data_target, data_label)
 
 
-def get_num_shards(num_samples: int, samples_per_shard: int, min_num_shards: int) -> int:
+def check_score_file(num_samples: int, curriculum_score_file: str):
```

Comment:
not sure about this function. All these checks can be interleaved with the actual logic when reading the file. This would avoid reading it in twice.
For example: python would complain anyway if you would try to call `int()` on something that cannot be converted to int, so no need to check for that in a separate loop.